core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -183,97 +168,156 @@ trait EntityComponent {
+      entityAttrRec <- entityAttributeQuery if entityAttrRec.ownerId === entityId && ! entityAttrRec.deleted
+    } yield entityAttrRec
+
+    // queries which may include "deleted" hidden entities
+
+    def findEntityByName(workspaceId: UUID, entityType: String, entityName: String): EntityQuery = {
+      filter(entRec => entRec.name === entityName && entRec.entityType === entityType && entRec.workspaceId === workspaceId)
+    }
+
     def findEntityById(id: Long): EntityQuery = {
       filter(_.id === id)
     }
 
-    def findEntityByIdAndVersion(id: Long, version: Long): EntityQuery = {
+    private def findEntityByIdAndVersion(id: Long, version: Long): EntityQuery = {
       filter(rec => rec.id === id && rec.version === version)
     }
 
-    // includes "deleted" hidden entities when using their hidden names
-    def lookupEntitiesByNames(workspaceId: UUID, entities: Set[AttributeEntityReference]): ReadAction[Seq[EntityRecord]] = {
-      if (entities.isEmpty) {
-        DBIO.successful(Seq.empty)
-      } else {
-        // slick can't do a query with '(entityType, entityName) in ((?, ?), (?, ?), ...)' so we need raw sql
-        EntityRecordRawSqlQuery.action(workspaceId, entities)
-      }
-    }
+    // Actions
+
+    // get a specific entity or set of entities: may include "hidden" deleted entities if not named "active"
 
-    // includes "deleted" hidden entities when using their hidden names
     def get(workspaceContext: SlickWorkspaceContext, entityType: String, entityName: String): ReadAction[Option[Entity]] = {
-      unmarshalEntities(EntityAndAttributesRawSqlQuery.actionForTypeName(workspaceContext, entityType, entityName)).map(_.headOption)
+      EntityAndAttributesRawSqlQuery.actionForTypeName(workspaceContext, entityType, entityName) map unmarshalEntities map(_.headOption)
     }
 
-    /**
-     * converts a query resulting in a number of records representing many entities with many attributes, some of them references
-     */
-    def unmarshalEntities(entityAndAttributesQuery: EntityQueryWithAttributesAndRefs): ReadAction[Iterable[Entity]] = {
-      entityAndAttributesQuery.result map { entityAttributeRecords =>
-        val attributeRecords = entityAttributeRecords.collect { case (entityRec, Some((attributeRec, referenceOption))) => ((entityRec.id, attributeRec), referenceOption) }
-        val attributesByEntityId = entityAttributeQuery.unmarshalAttributes(attributeRecords)
-
-        entityAttributeRecords.map { case (entityRec, _) =>
-          unmarshalEntity(entityRec, attributesByEntityId.getOrElse(entityRec.id, Map.empty))
-        }
-      }
+    def getEntities(entityIds: Traversable[Long]): ReadAction[Seq[(Long, Entity)]] = {
+      EntityAndAttributesRawSqlQuery.actionForIds(entityIds.toSet) map unmarshalEntitiesWithIds
     }
 
-    def unmarshalEntities(entityAttributeAction: ReadAction[Seq[EntityAndAttributesRawSqlQuery.EntityAndAttributesResult]]): ReadAction[Iterable[Entity]] = {
```

Comment:
Our Marshal/Unmarshal methods should always be independent of the related Slick actions.